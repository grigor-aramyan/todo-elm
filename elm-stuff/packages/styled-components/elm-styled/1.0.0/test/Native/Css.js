const { insert, flush } = require('../../src/Native/Css.js')

function getInsertedCss(index = 0) {
	const stylesheets = Array.from(document.querySelectorAll('style'))

	const stylesheet = stylesheets[index]

	if (!stylesheet) {
		return ''
	}

	return stylesheet.sheet.cssRules
		.map(cssRule => cssRule.cssText)
		.join('\n')
}

beforeEach(flush)

describe('inserting styles', () => {
	it('should insert a rule', () => {
		insert('.header { font-size: 2rem; }')
		insert('.text { font-size: 1.5rem; }')

		const insertedCSS = getInsertedCss()
		expect(insertedCSS).toMatchSnapshot()
	})

	it('should use a second stylesheet for many rules', () => {
		for (let i = 0; i < 65010; i++) {
			insert(`.class-${i} { font-size: 2rem; }`)
		}

		const insertedCSS = getInsertedCss(1)
		expect(insertedCSS).toMatchSnapshot()
	})

	it('should insert @import rules at the top', () => {
		insert('.footer { font-size: 2rem; }')
		insert('@import url("https://example.org");')
		insert('.box { font-size: 2rem; }')

		const insertedCSS = getInsertedCss()
		expect(insertedCSS).toMatchSnapshot()
	})
})
