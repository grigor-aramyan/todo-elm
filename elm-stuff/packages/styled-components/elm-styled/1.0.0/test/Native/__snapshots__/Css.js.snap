// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inserting styles should insert @import rules at the top 1`] = `
"@import url(https://example.org);
.footer {font-size: 2rem;}
.box {font-size: 2rem;}"
`;

exports[`inserting styles should insert a rule 1`] = `
".header {font-size: 2rem;}
.text {font-size: 1.5rem;}"
`;

exports[`inserting styles should use a second stylesheet for many rules 1`] = `
".class-65000 {font-size: 2rem;}
.class-65001 {font-size: 2rem;}
.class-65002 {font-size: 2rem;}
.class-65003 {font-size: 2rem;}
.class-65004 {font-size: 2rem;}
.class-65005 {font-size: 2rem;}
.class-65006 {font-size: 2rem;}
.class-65007 {font-size: 2rem;}
.class-65008 {font-size: 2rem;}
.class-65009 {font-size: 2rem;}"
`;
